script.py

```diff
@@ -0,0 +1,186 @@
+parser.add_argument('-u', '--unichem', help='Path to unichem src1src3.txt file', default=None, metavar='unichem_file_path', type=str, required=True)
+parser.add_argument('-e', '--ensembl', help='Path to pdb_chain_ensembl.csv file', default=None, metavar='ensembl_file_path', type=str, required=True)
+
+args = parser.parse_args()
+
+
+def get_structure(pdb_id):
+    """
+    Scrap the list of structures in complex with each molecule from PDBe API
+    Function to apply to a df column
+    --------------
+    Keyword arguments
+        pdb_id: id of the molecule for whom we want to know the structures associated
+    --------------
+    Return
+        data[pdb_id] : data is a dictionary, we want the value which is the list of structures
+    """
+    url = f'https://www.ebi.ac.uk/pdbe/api/pdb/compound/in_pdb/{pdb_id}'
+    response = requests.get(url)
+    try:
+        data = response.json()
+        return data[pdb_id]
+    except:
+        if len(response.json()) == 0:
+            return []
+
+
+def create_pdb_target_gene_df(path_id_file, unichem_molecule_struct_spark_df):
+
+    pdb_chain_ensembl = spark.read.csv(path_id_file, sep=',', header=True, comment='#')
+    pdb_chain_ensembl = pdb_chain_ensembl.select('PDB', 'GENE_ID').distinct()
+
+    # print('----- STRUCTURE & TARGET -----')
+    # pdb_chain_ensembl.show()
+
+    exploded_df = (unichem_molecule_struct_spark_df
+                   .select('MOLECULE_PDB_ID',
+                           'MOLECULE_CHEMBL_ID',
+                           f.explode(unichem_molecule_struct_spark_df.STRUCTURE_ID)))
+
+    # print('----- STRUCTURE & TARGET EXPLODED -----')
+    # exploded_df.show()
+
+    chain_ensembl_struct_mol_joined = (pdb_chain_ensembl
+                                       .join(exploded_df, pdb_chain_ensembl["PDB"] == exploded_df["col"])
+                                       .drop('col'))
+
+    return chain_ensembl_struct_mol_joined
+
+
+def main():
+
+    # TIMER 1 START
+    start_1 = timeit.default_timer()
+
+    # ----- GET MOLECULE ID -----
+    # Molecule DataFrame
+    molecule_df = (
+        spark.read
+        .parquet(args.parquet)
+        .select(f.col('id'))
+        .withColumnRenamed('id', 'MOLECULE_CHEMBL_ID')
+    )
+
+    # ----- GET PDB ID -----
+    # Unichem molecules DataFrame
+    unichem_df = spark.read.csv(args.unichem, sep=r'\t', header=True)
+
+    # Join Unichem and Molecule DataFrame
+    unichem_molecule_df = (molecule_df
+                           .join(unichem_df, unichem_df["From src:'1'"] == molecule_df['MOLECULE_CHEMBL_ID'])
+                           .withColumnRenamed("To src:'3'", 'MOLECULE_PDB_ID')
+                           .drop("From src:'1'")
+                           )
+
+    # TIMER 2 START
+    start_2 = timeit.default_timer()
+
+    # ----- GET STRUCTURE ID -----
+    # Apply get_structure function on the Unichem-Molecule DataFrame
+
+    unichem_molecule_struct_pd_df = (unichem_molecule_df
```

Comment:
Yes, using a function where you requests data from an API, it makes sense to convert it to pandas dataframe. I like to see the `.assign` method being used. It can be further optimized a bit, but it's good.